src/core/announcements/crypto.ts

```diff
@@ -0,0 +1,39 @@
+import { ConfigOpts, requireGetSigner } from "../../config";
+import { HexString } from "../../types/Strings";
+import { ethers } from "ethers";
+import { Announcement, AnnouncementWithSignature } from "./announcementTypes";
+import { serialize } from "./serialization";
+
```

Comment:
Let's put the signed message types in this file too, i.e. `SignedAnnouncement`, etc.